### ISSUE TITLE
Unable to deploy kubernetes-ingress >= 1.17.6 against kubernetes v1.17.x

### ISSUE THREAD

**Issue by flabatut** [haproxytech/helm-charts#111]
# Description

* Starting with kubernetes-ingress 1.17.6, when trying to deploy release against kubernetes v1.17.x, the following exception occurs:

```
  Error: Failed to render chart: exit status 1: Error: unable to build kubernetes objects from release manifest: unable to recognise "": no matches for kind "IngressClass" in version "networking.k8s.io/v1"
```

* IngressClass is only available starting with kubernetes v1.18: https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/

* Prior to kubernetes-ingress 1.17.6, appropriate condition was present in controller-ingressclass.yaml template: https://github.com/haproxytech/helm-charts/blame/5f24cb8048c908d037efbde74e78b633146d2c2f/kubernetes-ingress/templates/controller-ingressclass.yaml

* Sounds like there is no longer way to toggle ingressClass resource addition. Is it intended or can we get the condition back ?

# Versions

kubernetes: v1.17.16
helm: v3.5.4

**Comment by dkorunic** [comment]
@flabatut Please make sure to use Helm Chart `1.17.9`, as a few problems in regards to Helm/Chart-testing/ArgoCD have ben identified in 1.17.x releases prior to `1.17.9`.
On a related matter, `Capabilities.KubeVersion.GitVersion` has been deprecated in all recent Helm v3 releases. Checking capabilities with `Capabilities.ApiVersions.Has` is now a recommended way and it looks like in your case that Helm is claiming that `Capabilities.ApiVersions` has `networking.k8s.io/v1/IngressClass`, but when template is being rendered using that capability results in an error. Moving forward, can you please:
- make sure to upgrade to Helm Chart [1.17.9](https://github.com/haproxytech/helm-charts/releases/tag/kubernetes-ingress-1.17.9),
- upgrade Helm to [3.7.1](https://github.com/helm/helm/releases/tag/v3.7.1) (or 3.6.3 at least) and report if the helm install/upgrade is still failing.

Also note that K8s `1.17` is very old at this point (EOL-ed on **25 Dec 2020**) and we plan to completely stop supporting K8s older than 1.18 by EOY.

**Comment by flabatut** [comment]
@dkorunic already tried with `1.17.9` (tried with that one at first actually), same concern unfortunately

**Comment by dkorunic** [comment]
@flabatut I can confirm that we've been able to reproduce and a fix is incoming.

**Comment by dkorunic** [comment]
@flabatut I can confirm that we've been able to reproduce and a fix is incoming.

**Comment by dkorunic** [comment]
@flabatut Fixed in relase [1.17.10](https://github.com/haproxytech/helm-charts/releases/tag/kubernetes-ingress-1.17.10) and it is compatible with K8s 1.17 for now. However the previous note still stands, 1.17 has been EOLed at this point and we don't expect to continue supporting it after EOY.
